CRVResponse/src/CrvPhotonGenerator_module.cc

```diff
@@ -209,32 +298,54 @@ namespace mu2e
+            double sectorScintillationYield=_scintillationYields[CRVSectorNumber];
+            double adjustedYield=0;
             do
             {
-              adjustment=_randGaussQ.fire(0, _scintillationYields[CRVSectorNumber]*_scintillationYieldVariation);
-            } while(adjustment<-_scintillationYields[CRVSectorNumber]*_scintillationYieldVariationCutoff);
-            _scintillationYieldAdjustments[step.barIndex()] = adjustment;
+              adjustedYield=_randGaussQ.fire(sectorScintillationYield, sectorScintillationYield*_scintillationYieldVariation);
+            } while(adjustedYield<sectorScintillationYield*_scintillationYieldVariationCutoffLow ||
+                    adjustedYield>sectorScintillationYield*_scintillationYieldVariationCutoffHigh);
+
+            _scintillationYieldsAdjusted[step.barIndex()] = adjustedYield;
           }
-          double scintillationYieldAdjustment = _scintillationYieldAdjustments[step.barIndex()];
+          double currentAdjustedYield = _scintillationYieldsAdjusted[step.barIndex()];
 
           boost::shared_ptr<mu2eCrv::MakeCrvPhotons> &photonMaker=_makeCrvPhotons.at(CRVSectorNumber);
+          photonMaker->SetScintillationYield(currentAdjustedYield);
           photonMaker->MakePhotons(pos1Local, pos2Local, t1, t2,
                                         avgBeta, charge,
                                         step.visibleEDep(),
                                         step.pathLength(),
-                                        scintillationYieldAdjustment,
-                                        _lookupTableReflectors[CRVSectorNumber]);
+                                        _reflectors[CRVSectorNumber]);
 
           art::Ptr<CrvStep> crvStepPtr(CrvSteps,istep);
           for(int SiPM=0; SiPM<4; ++SiPM)
           {
             std::pair<CRSScintillatorBarIndex,int> barIndexSiPMNumber(step.barIndex(),SiPM);
-            std::vector<CrvPhotons::SinglePhoton> &photons = photonMap[barIndexSiPMNumber];
             const std::vector<double> &times=photonMaker->GetArrivalTimes(SiPM);
-            for(size_t itime=0; itime<times.size(); ++itime) photons.emplace_back(times[itime],crvStepPtr);
+            if(times.empty()) continue;
+            std::vector<CrvPhotons::SinglePhoton> &photons = photonMap[barIndexSiPMNumber];
+            for(size_t itime=0; itime<times.size(); ++itime)
+            {
+              double timeTmp=times[itime];
+              if(spillType==EventWindowMarker::SpillType::onspill)
+              {
+                timeTmp = fmod(timeTmp,_microBunchPeriod);
+                //photons before the digitization start get removed except photons 
+                //in the first 55ns which get moved to the interval between the end of 
+                //the microbunch period and the digitization end
+                if(timeTmp<_digitizationEnd-_microBunchPeriod) timeTmp+=_microBunchPeriod;  
+                if(timeTmp<_digitizationStart) continue;
```

Comment:
I think you need to be careful here - this is all in times wrt protons I believe and so doesn't have any information about the jitter in the event marker time - you will lose the photons that get moved back into the digitization window by the jitter

Reply:
I added the jitter to the digitization window.